src/utilities/generic.ts

```diff
@@ -46,6 +46,15 @@ export function getTarget(event: Event) {
   }
 }
 
+/**
+ * Returns true if a value is considered "empty". False otherwise.
+ */
+export function isEmpty(value: any): boolean {
+  return (
+    value === null || value === undefined || value === '' || Number.isNaN(value)
```

Comment:
Is this a common TS idiom? I'll need to learn the difference between `null` and `undefined`, sounds like this function should be a TS builtin.

Reply:
Yeah [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null) and [undefined](https://developer.mozilla.org/en-US/docs/Glossary/undefined) are both different JavaScript primitives generally representing the absence of a value.